src/shaders/background_pattern.vertex.glsl

```diff
@@ -12,9 +12,17 @@ attribute vec2 a_pos;
 varying vec2 v_pos_a;
 varying vec2 v_pos_b;
 
+#if defined( FOG ) && !defined( RENDER_TO_TEXTURE )
+varying vec3 v_fog_pos;
+#endif
```

Comment:
It can be, but since the unused varying still incurs some cost (I checked and the answer was: stop guessing what the compiler does and remove unused code), I opted to keep it in individual shaders so that each can make its own choice. For example, some shaders may not have any fog-specific behavior. Rather than carefully cutting out FOG from their defines (which we could do; do we do this?), this only defines the varying if it's used.

Reply:
I've kept it as-is for now. I'm not against moving it, but would prefer to move that to a subsequent PR if we decide to go for it.